### PR TITLE
🏉 Honor is_active when creating contact fields

### PR DESCRIPTION
**Pull request by dodobas**
Fixes: https://github.com/rapidpro/rapidpro/issues/916

### CODE REVIEW

File: temba/contacts/models.py [hunk 1]
```diff
@@ -504,12 +510,6 @@ def get_or_create(cls, org, user, key, label=None, show_in_table=None, value_typ
                 update_events = False
                 changed = False
 
-                # make this as active
```
Comment (dodobas) [inline temba/contacts/models.py]: looks like we were toggling `is_active` on hidden fields on `get_or_create`, not sure why was this required, anyone remembers?

File: temba/contacts/models.py [hunk 2]
```diff
@@ -459,7 +465,7 @@ def is_valid_label(cls, label):
 
     @classmethod
     def hide_field(cls, org, user, key):
-        existing = ContactField.user_fields.collect_usage().filter(org=org, key=key).first()
+        existing = ContactField.user_fields.collect_usage().active_for_org(org=org).filter(key=key).first()
```
Comment (dodobas) [inline temba/contacts/models.py]: hiding a hidden field is now a *noop*

File: temba/campaigns/tests.py [hunk 3]
```diff
@@ -1613,28 +1613,6 @@ def test_scheduling(self):
         self.assertEqual(planting_reminder, fire.event)
         self.assertEqual(9, fire.scheduled.day)
 
-        # "hide" planting reminder Campaign, in order to hide the "planting_date" field
```
Comment (dodobas) [inline temba/campaigns/tests.py]: this test is not relevant anymore, because users can't remove fields that are used in campaigns

File: temba/contacts/models.py (hunk 1 above)
Comment (dodobas) [reply temba/contacts/models.py]: As this was removed, now there is missing coverage in `campaigns/models.py` (https://codecov.io/gh/nyaruka/rapidpro/compare/02839c618f54eab08fe3e0326fa7d073b7cd3479...c19a9b350bc1107ea4b6262f9c04d2305635e41b/changes) 

If field was deleted then remove all EventFires, and if field is still active then also create new EventFires for that field, AFAICT we don't actually need this code anymore since users cant remove a contact field with uses, and user can't 'reactivate' a deleted field

if there are no objections i will remove it

Comment (rowanseymour) [reply temba/contacts/models.py]: Well the previous logic was to re-activate a field with the same key. The new logic is to ignore inactive fields with same key and create a new one - which does feel more right. If a thing is deleted it's deleted.

We also used to delete EventFires when the field was being deleted.. I think now the EventFires are left and just ignored when they fire - allowing field deletion to be more much more performant.

Comment (nicpottier) [reply temba/contacts/models.py]: Since we do no cleanup on contact field values when a contact field is removed (IE, the JSON on the contact doesn't change) it is indeed important we don't resuscitate old contact fields or they will still have values on the contact. So yes, we should never reuse contact fields.
